commit 6a1f0ebf35ffaa6899bcb95ab022fc4ae3c47766 (refs/changes/58/3858/4)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2018-04-25T19:08:25+00:00 (1 year, 5 months ago)
    
    TRITON-326 NAPI should dump core when an uncaught error is thrown
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/napi.js b/lib/napi.js
index ab237ae..3adac09 100644
--- a/lib/napi.js
+++ b/lib/napi.js
@@ -32,10 +32,7 @@ var os = require('os');
 var restify = require('restify');
 var trace_event = require('trace-event');
 var util = require('util');
-var verror = require('verror');
-
-var VError = verror.VError;
-var WError = verror.WError;
+var VError = require('verror');
 
 
 // --- Globals
@@ -117,6 +114,7 @@ function NAPI(opts) {
     var server = this.server = restify.createServer({
         log: opts.log,
         name: PKG.description,
+        handleUncaughtExceptions: false,
         version: PKG.version
     });
 
@@ -183,16 +181,6 @@ function NAPI(opts) {
         })(req, res, route, err);
     });
 
-    server.on('uncaughtException', function (req, res, route, err) {
-        res.send(new WError(err, 'Internal error'));
-        restify.auditLogger({
-            log: req.log.child({
-                component: 'audit',
-                route: route && route.name
-            }, true)
-        })(req, res, route, err);
-    });
-
     endpoints.registerEndpoints(server, before, this.log);
 
     mod_mooremachine.FSM.call(this, 'waiting');
diff --git a/package.json b/package.json
index 842b1d2..afc80e4 100644
--- a/package.json
+++ b/package.json
@@ -19,7 +19,7 @@
     "moray": "3.4.2",
     "node-uuid": "1.4.7",
     "portolan-moray": "git+https://github.com/joyent/sdc-portolan-moray.git#7e2c4ca",
-    "restify": "4.1.1",
+    "restify": "4.3.0",
     "restify-warden": "1.1.0",
     "sdc-clients": "9.2.0",
     "tape": "4.5.1",
